src/python/pants/backend/nfpm/config.py

```diff
@@ -0,0 +1,66 @@
+# Copyright 2023 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import annotations
+
+from typing import TypedDict
+
+import yaml
+
+from pants.backend.nfpm.fields.contents import (
+    NfpmContentFileGroupField,
+    NfpmContentFileModeField,
+    NfpmContentFileMtimeField,
+    NfpmContentFileOwnerField,
+)
+from pants.engine.target import Target
+
+
+class OctalInt(int):
+    # noinspection PyUnusedLocal
+    @staticmethod
+    def represent_octal(dumper: yaml.representer.BaseRepresenter, data: int) -> yaml.Node:
+        # YAML 1.2 octal: 0o7777 (py: f"0o{data:o}" or f"{data:#o}" or oct(data))
+        # YAML 1.1 octal: 07777 (py: f"0{data:o}")
+        # Both octal reprs are supported by `gopkg.in/yaml.v3` which parses YAML in nFPM.
+        # See: https://github.com/go-yaml/yaml/tree/v3.0.1#compatibility
+        # PyYAML only supports reading YAML 1.1, so we use that.
+        return yaml.ScalarNode("tag:yaml.org,2002:int", f"0{data:o}")
+
+
+# This is an unfortunate import-time side effect: PyYAML does registration globally.
+yaml.add_representer(OctalInt, OctalInt.represent_octal, Dumper=yaml.SafeDumper)
```

Comment:
Apparently it is just shorthand for calling `add_representer` on the `Dumper` class parameter.  You probably can just subclass `SafeDumnper` and call its `add_representer` class method directly?

Reply:
see https://github.com/yaml/pyyaml/blob/69c141adcf805c5ebdc9ba519927642ee5c7f639/lib/yaml/__init__.py#L336